### PR TITLE
Feat: Split fetching of UTXOs of an address to separate endpoints

### PR DESCRIPTION
**Pull request by abbasabid**
# Description of change

- Added separate endpoints for basic, alias and nft outputs by address and return outputResponse for respective outputIds.
- Use these new endpoints on AddressPage.

Resolves https://github.com/iotaledger/explorer/issues/645

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/services/stardust/stardustTangleCacheService.ts [hunk 1]
```diff
@@ -254,6 +255,87 @@ export class StardustTangleCacheService extends TangleCacheService {
...
+     * Get the basic output details for an address.
+     * @param networkId The network in context.
+     * @param address The address in bech32 format.
+     * @returns The basic output details.
+     */
+    public async basicOutputsDetails(
+        networkId: string,
+        address: string
+    ): Promise<IAddressDetailsResponse | undefined> {
+        if (!this._stardustSearchCache[networkId][`${address}-basic-outputs-details`]?.data?.outputs) {
```
Comment (msarcev) [inline client/src/services/stardust/stardustTangleCacheService.ts:268]: Extract the "built key" to a `const`, please. For all three methods

**Review by msarcev — CHANGES_REQUESTED** [review]

**Comment by msarcev** [comment]
Looks good mostly. Please, also remove the old route from the whole stack (the one that fetched outputs in bundle)

**Review by brancoder — CHANGES_REQUESTED** [review]

File: client/src/services/stardust/stardustTangleCacheService.ts [hunk 2]
```diff
@@ -237,21 +237,93 @@ export class StardustTangleCacheService extends TangleCacheService {
...
 
-        return response;
+        if (!cacheEntry) {
+            const response = await this._api.basicOutputsDetails({
+                network: networkId,
+                address
+            });
+
+            if (response.outputs) {
+                this._stardustSearchCache[networkId][`${address}-basic-outputs-details`] = {
```
[the remark this answers is not in the document]
Comment (abbasabid) [reply client/src/services/stardust/stardustTangleCacheService.ts]: Done.

**Review by brancoder — APPROVED** [review]
LGTM

File: client/src/app/routes/stardust/AddressPage.tsx [hunk 3]
```diff
@@ -95,8 +98,10 @@ const AddressPage: React.FC<RouteComponentProps<AddressRouteProps>> = (
 
     useEffect(() => {
         if (bech32AddressDetails) {
-            // eslint-disable-next-line no-void
-            void loadAddressData();
+            void getAddressBalance();
+            void loadBasicAddressData();
+            void loadAliasAddressData();
+            void loadNftAddressData();
```
Comment (msarcev) [inline client/src/app/routes/stardust/AddressPage.tsx:104]: Naming for these methods above sounds a bit misleading, just call it what it is... `loadAddressBasicOutputs`, `loadAddressAliasOutputs`

**Review by msarcev — APPROVED** [review]

**Pull request merged**